packages/alea-frontend/components/QuizDashboard.tsx

```diff
@@ -253,34 +261,67 @@ const QuizDashboard: NextPage<QuizDashboardProps> = ({ courseId, quizId, onQuizI
   }, [selectedQuizId, courseId, currentTerm, quizzes]);
 
   useEffect(() => {
-    if (selectedQuizId === NEW_QUIZ_ID) {
+    if (selectedQuizId !== NEW_QUIZ_ID) return;
+    const now = Date.now();
+    const upcomingLecture = lectureSchedule
+      .filter((lec) => lec.hasQuiz)
+      .map((lec) => {
+        const [sh, sm] = lec.lectureStartTime.split(':').map(Number);
+        const weekdayIndex = [
+          'Sunday',
+          'Monday',
+          'Tuesday',
+          'Wednesday',
+          'Thursday',
+          'Friday',
+          'Saturday',
+        ].indexOf(lec.lectureDay);
+        let lectureDate = dayjs().day(weekdayIndex);
+        if (lectureDate.isBefore(dayjs(), 'day')) {
```

Comment:
You can't just check before/after based on day. it has to be time based and consider timezone.
 
Consider this case: the quiz is scheduled for `Tuesday Dec 2 at 16:00` and the prof creates a quiz at 08:00 on the same day.

`lectureDate` would be `Dec 2 Tuesday 00:00` and then this code would unnecessarily add a week making it `Dec 9`